app/controllers/api/v1/incident_contacts_controller.rb

```diff
@@ -18,6 +18,26 @@ def assign_contact_to_incident
     end
   end
 
+  def reassign_role_to_different_contact
+    incident = Incident.where(id: incident_contact_params[:incident_id])
+    contact = Contact.where(id: incident_contact_params[:contact_id])
+    if incident.empty?
+      render json: {error: "Incident does not exist"}, status: :not_found
+    elsif contact.empty?
+      render json: {error: "Contact does not exist"}, status: :not_found
```

Comment:
Love the error handling, if we have the time, maybe refactor into a error serializer or controller